documentation/asciidoc/computers/configuration/device-tree.adoc

```diff
@@ -991,3 +991,121 @@ Further ports can be controlled by the DT, for more details see <<part3,section
 ==== Further help
 
 If you've read through this document and not found the answer to a Device Tree problem, there is help available. The author can usually be found on Raspberry Pi forums, particularly the https://forums.raspberrypi.com/viewforum.php?f=107[Device Tree] forum.
+
+[[part5.0]]
+=== Firmware parameters
+The firmware uses the special https://www.kernel.org/doc/html/latest/devicetree/usage-model.html#runtime-configuration[chosen] node to pass parameters between the bootloader and/or firmware and the operating system.
+
+`boot-mode` - 32-bit integer
+
+The boot-mode used to load the kernel. See xref:raspberry-pi.adoc#BOOT_ORDER[BOOT_ORDER].
+
+`overlay_prefix` - string
+
+The xref:config_txt.adoc#overlay_prefix[overlay_prefix] string selected by `config.txt`.
+
+`os_prefix` - string
+
+The xref:config_txt.adoc#os_prefix[os_prefix] string selected by `config.txt`.
+
+`partition` - 32-bit integer
+
+The partition number used during boot.
+
+`pm_rsts` - 32-bit integer
+
+The value of the `PM_RSTS` register during boot.
+
+`rpi-boardrev-ext` - 32-bit integer
+
+The extended board revision code from xref:raspberry-pi.adoc#otp-register-and-bit-definitions[OTP row 33].
+
+`rpi-country-code` - 32-bit integer
+
+The country code used used by https://github.com/raspberrypi-ui/piwiz[PiWiz] - Pi400 only.
+
+==== BCM2711 bootloader properties `/chosen/bootloader`
+The following properties are specific to BCM2711 SPI EEPROM bootloader.
+
+`build_timestamp` - 32-bit integer
+
+The UTC build time for the EEPROM bootloader.
+
+`capabilities` - 32-bit integer
+
+This bit field which describes the features supported by the current bootloader. This may be used to check whether a feature (e.g. USB boot) is supported before enabling it in the bootloader EEPROM config.
```

Comment:
"This bit field which describes the features" -> "This bit field describes the features"

Reply:
Or, "This bit field, which describes the features supported by the current bootloader, may be used to check whether a feature (e.g. USB boot) is supported before enabling it in the bootloader EEPROM config."

Reply:
Ah, the lack of comma caused me to mis-parse this originally :wink: